Handshake is broken for NtC

It looks like it's trying to treat the version data as if it was the NtN format.

```
$ ./go-ouroboros-network -preview -address localhost:8082 local-tx-submission -tx-file README.md 
panic: interface conversion: interface {} is uint64, not []interface {}

goroutine 11 [running]:
github.com/cloudstruct/go-ouroboros-network/protocol/handshake.(*Client).handleAcceptVersion(0x0?, {0x6cd9e8?, 0xc00019e000?})
        /home/agaffney/dev/cloudstruct/go-ouroboros-network/protocol/handshake/client.go:71 +0xd7
github.com/cloudstruct/go-ouroboros-network/protocol/handshake.(*Client).handleMessage(0xb400000000000003?, {0x6cd9e8, 0xc00019e000}, 0xc8?)
        /home/agaffney/dev/cloudstruct/go-ouroboros-network/protocol/handshake/client.go:56 +0xdc
github.com/cloudstruct/go-ouroboros-network/protocol.(*Protocol).handleMessage(0xc00007d450, {0x6cd9e8, 0xc00019e000}, 0x40?)
        /home/agaffney/dev/cloudstruct/go-ouroboros-network/protocol/protocol.go:351 +0x1c8
github.com/cloudstruct/go-ouroboros-network/protocol.(*Protocol).recvLoop(0xc00007d450)
        /home/agaffney/dev/cloudstruct/go-ouroboros-network/protocol/protocol.go:280 +0x3b5
created by github.com/cloudstruct/go-ouroboros-network/protocol.(*Protocol).Start
        /home/agaffney/dev/cloudstruct/go-ouroboros-network/protocol/protocol.go:95 +0x2ea
```

--- a/protocol/handshake/client.go
+++ b/protocol/handshake/client.go
@@ -68,10 +68,12 @@ func (c *Client) handleAcceptVersion(msgGeneric protocol.Message) error {
 	}
 	msg := msgGeneric.(*MsgAcceptVersion)
 	fullDuplex := false
-	versionData := msg.VersionData.([]interface{})
-	//nolint:gosimple
-	if versionData[1].(bool) == DIFFUSION_MODE_INITIATOR_AND_RESPONDER {
-		fullDuplex = true
+	if c.Mode() == protocol.ProtocolModeNodeToNode {
+		versionData := msg.VersionData.([]interface{})
+		//nolint:gosimple
+		if versionData[1].(bool) == DIFFUSION_MODE_INITIATOR_AND_RESPONDER {
+			fullDuplex = true
+		}
 	}
 	return c.config.FinishedFunc(msg.Version, fullDuplex)
 }

